File: javascript/hamming/hamming.js
var Hamming = function() {
  this.compute = function(firstStrand, secondStrand) {
    if (firstStrand.length != secondStrand.length) {
      throw new Error('DNA strands must be of equal length.');
    }

    var hammingDistance = 0;
    for (let i = 0; i < firstStrand.length; i++) {
      if (firstStrand[i] != secondStrand[i]) {
        hammingDistance++;
      }
    }

    return hammingDistance;
  };
};

module.exports = Hamming;

File: ruby/hamming/hamming.rb
module BookKeeping
  VERSION = 3
end

class Hamming
  def self.compute(first_strand, second_strand)
    throw ArgumentError unless first_strand.length == second_strand.length

    return 0 if first_strand == second_strand

    return (0...first_strand.length).count do |index|
      first_strand[index] != second_strand[index]
    end
  end
end
